### ISSUE TITLE
Introduces bangs in invalid places

### ISSUE THREAD

**Issue by nomeata** [Genetic-Strictness/Autobahn#8]
I get: 

```
src/Stats.hs:308:29:
    Parse error in pattern: !~ ((! Calculations {..}))
```

I guess a bang before a `~` is not allowed. Well, the genetic algorithm will hopefully figure that out :-)

**Comment by dnunez02** [comment]
Correct. Autobahn is supposed to give this one the worst possible score. As you found, it did not.

See the comment in #9 for a bit more info.

**Comment by nomeata** [comment]
Shouldn’t it in this case undo the `~(…)` instead of adding a `!`?

**Comment by dnunez02** [comment]
I see. You are right.
